primitives/account/src/staking_contract/traits.rs

```diff
@@ -337,36 +342,50 @@ impl AccountTransactionInteraction for StakingContract {
         // Parse transaction data.
         let data = OutgoingStakingTransactionProof::parse(transaction)?;
 
-        let receipt = match data {
+        let mut acc_info: AccountInfo;
+
+        match data {
             OutgoingStakingTransactionProof::DeleteValidator { proof } => {
                 // Get the validator address from the proof.
                 let validator_address = proof.compute_signer();
 
-                Some(
-                    StakingContract::delete_validator(
-                        accounts_tree,
-                        db_txn,
-                        &validator_address,
-                        block_height,
-                    )?
-                    .serialize_to_vec(),
-                )
+                acc_info = StakingContract::delete_validator(
+                    accounts_tree,
+                    db_txn,
+                    &validator_address,
+                    block_height,
+                )?
+                .into();
             }
             OutgoingStakingTransactionProof::Unstake { proof } => {
                 // Get the staker address from the proof.
                 let staker_address = proof.compute_signer();
 
-                StakingContract::unstake(
+                acc_info = StakingContract::unstake(
                     accounts_tree,
                     db_txn,
                     &staker_address,
                     transaction.total_value(),
                 )?
-                .map(|r| r.serialize_to_vec())
+                .into();
             }
         };
-
-        Ok(receipt)
+        acc_info.logs.insert(
```

Comment:
Does the order of logs matter here or can this also be
```suggestion
        acc_info.logs.push(
```
?

If the order does matter, add a comment explaining why.